test/InterfaceStub.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/** @file InterfaceStub.cpp
+ * @author Marek Kotewicz <marek@ethdev.com>
+ * @date 2015
+ */
+
+#include <boost/test/unit_test.hpp>
+#include <libdevcore/CommonJS.h>
+#include "TestUtils.h"
+
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+using namespace dev::test;
+
+BOOST_FIXTURE_TEST_SUITE(InterfaceStub, ParallelInterfaceStubFixture)
+
+BOOST_AUTO_TEST_CASE(blocks)
+{
+	enumerateInterfaces([](Json::Value const& _json, dev::eth::InterfaceStub& _client) -> void
+	{
+		for (string const& name: _json["postState"].getMemberNames())
+		{
+			Json::Value o = _json["postState"][name];
+			Address address(name);
+
+			// balanceAt
+			u256 expectedBalance = u256(o["balance"].asString());
+			u256 balance = _client.balanceAt(address);
+			BOOST_CHECK_EQUAL(expectedBalance, balance);
```

Comment:
unfortunately `BOOST_CHECK_EQUAL` is not thread safe due to things described [here](https://codecrafter.wordpress.com/2012/11/01/c-unit-test-framework-adapter-part-3/). I have only tested this code on mac os. This caused various glitches in console output.